e2etests/test/eventhub_receiver_helper.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var eventHubClient = require('azure-event-hubs').Client;
+var util = require('util');
+var EventEmitter = require('events');
+var closeDeviceEventHubClients = require('./testUtils.js').closeDeviceEventHubClients;
+
+var hubConnectionString = process.env.IOTHUB_CONNECTION_STRING;
+
+var EventHubReceiverHelper = function() {
+};
+util.inherits(EventHubReceiverHelper, EventEmitter);
+
+EventHubReceiverHelper.prototype.openClient = function(done) {
+  var self = this;
+  this.ehClient = eventHubClient.fromConnectionString(hubConnectionString);
+  this.ehReceivers = [];
+
+  this.ehClient.open()
+      .then(self.ehClient.getPartitionIds.bind(self.ehClient))
+      .then(function (partitionIds) {
+        return partitionIds.map(function (partitionId) {
+          return self.ehClient.createReceiver('$Default', partitionId,{ 'startAfterTime' : Date.now()}).then(function(receiver) {
+            self.ehReceivers.push(receiver);
+            receiver.on('errorReceived', function(err) {
+              self.emit(err);
+            });
+            receiver.on('message', function (eventData) {
+              self.emit('message', eventData);
+            });
+          });
+        });
+      })
+      .catch(function(err) {
+        done(err);
+      })
+      .then(function() {
+        done();
+      });
+};
+
+EventHubReceiverHelper.prototype.closeClient = function(done) {
+  closeDeviceEventHubClients(null, this.ehClient, this.ehReceivers, done);
```

Comment:
>closeDeviceEventHubClients [](start = 2, length = 26)

ehh... can we separate this in 2 and call the close just on the event hubs instead? i know it's just helper code, but this looks like uncontrolled growth of code :)